src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -233,6 +238,40 @@ private async Task RunLoop(CancellationToken cancellationToken)
                 }
             }
 
+            TimeSpan cancellationTime = TimeSpan.Parse(_options.Value.LogScrapingTimeout ?? "00:10:00");
+
+            try
+            {
+                _logger.LogInformation("Starting log scraping");
+
+                var logScrapingTimeoutCancellationTokenSource = new CancellationTokenSource(cancellationTime);
+                var logScrapingTimeoutCancellationToken = logScrapingTimeoutCancellationTokenSource.Token;
+
+                var combinedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, logScrapingTimeoutCancellationToken);
+                var combinedCancellationToken = combinedCancellationTokenSource.Token;
+
+                Stopwatch stopWatch = Stopwatch.StartNew();
+
+                await GetImageNames(records, combinedCancellationToken);
+
+                if (combinedCancellationToken.IsCancellationRequested)
+                {
+                    _logger.LogWarning($"Log scraping timed out after {cancellationTime}");
```

Comment:
I'm not sure how this will get hit, because GetImageNames will throw when the token is cancelled.

Reply:
I don't think it will, I forgot to change it when changing the other method, removed